vpr/src/device/rr_graph_view.h

```diff
@@ -96,6 +96,60 @@ class RRGraphView {
         return node_storage_.fan_in(node);
     }
 
+    /* Get the xlow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xlow(RRNodeId node) const {
+        return node_storage_.node_xlow(node);
+    }
+
+    /* Get the xhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xhigh(RRNodeId node) const {
+        return node_storage_.node_xhigh(node);
+    }
+
+    /* Get the ylow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_ylow(RRNodeId node) const {
+        return node_storage_.node_ylow(node);
+    }
+
+    /* Get the yhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_yhigh(RRNodeId node) const {
+        return node_storage_.node_yhigh(node);
+    }
+
+    /* Get the length of a routing resource node. This function is inlined for runtime optimization. */
+    inline int node_length(RRNodeId node) const {
+        return 1 + node_xhigh(node) - node_xlow(node) + node_yhigh(node) - node_ylow(node);
+    }
+
+    /* Check if routing resource node is initialized. This function is inlined for runtime optimization. */
+    inline bool node_is_initialized(RRNodeId node) const {
+        return !((node_type(node) == SOURCE)
+                 && (node_xlow(node) == 0) && (node_ylow(node) == 0)
+                 && (node_xhigh(node) == 0) && (node_yhigh(node) == 0));
+    }
+
+    /* Check if two routing resource nodes are adjacent. This function is inlined for runtime optimization. */
+    inline bool nodes_are_adjacent(RRNodeId first_node, RRNodeId second_node) const {
```

Comment:
- The definition of adjacent should be well explained here as a code comment. When developers read this API, they know when to use it and how to evaluate the return value
- You may not need to use local variables in this inline function, you can make it much compact.